Code/Framework/AzToolsFramework/AzToolsFramework/Script/LuaEditorSystemComponent.cpp

```diff
@@ -0,0 +1,236 @@
+                else
+                {
+                    AZ_Error(LogName, false, outcome.GetError().c_str());
+                }
+            };
+
+            creators.push_back({ "LuaComponent_creator", "Lua Component Script", QIcon(), luaComponentAssetCreator });
+        }
+
+        void LuaEditorSystemComponent::AddSourceFileOpeners(
+            const char* fullSourceFileName,
+            [[maybe_unused]] const AZ::Uuid& sourceUUID,
+            AzToolsFramework::AssetBrowser::SourceFileOpenerList& openers)
+        {
+            if (AZ::IO::Path(fullSourceFileName).Extension() == LuaExtension)
+            {
+                auto luaScriptOpener = [](const char* fullSourceFileNameInCallback, [[maybe_unused]] const AZ::Uuid&)
+                {
+                    AzToolsFramework::EditorRequestBus::Broadcast(
+                        &AzToolsFramework::EditorRequests::LaunchLuaEditor, fullSourceFileNameInCallback);
+                };
+
+                openers.push_back({ "O3DE_LUA_Editor", "Open in Open 3D Engine LUA Editor...", QIcon(), luaScriptOpener });
+            }
+        }
+
+        void LuaEditorSystemComponent::HandleInitialFilenameChange(const AZStd::string_view fullFilepath)
+        {
+            AZ::IO::Path filepath = AZ::IO::Path(fullFilepath);
+            if (filepath.Extension() == LuaExtension)
+            {
+                AZStd::string_view& filename = filepath.Stem().Native();
+                AZStd::string scriptBoilerplate = GenerateLuaComponentBoilerplate(filename);
+
+                auto outcome = SaveLuaScriptFile(fullFilepath, scriptBoilerplate);
+                if (!outcome.IsSuccess())
+                {
+                    AZ_Error(LogName, false, outcome.GetError().c_str());
+                }
+            }
+        }
+
+        AZStd::string LuaEditorSystemComponent::GenerateLuaComponentBoilerplate(const AZStd::string& componentName)
+        {
+            constexpr const char* namePlaceholder = "$SCRIPT_NAME";
+            AZStd::string luaComponentBoilerplate = R"LUA(-- $SCRIPT_NAME.lua
+
+local $SCRIPT_NAME = 
+{
+    Properties =
+    {
+        -- Property definitions
+    }
+}
+
+function $SCRIPT_NAME:OnActivate()
+    -- Activation Code
+end
+
+function $SCRIPT_NAME:OnDeactivate()
+    -- Deactivation Code
+end
+
+return $SCRIPT_NAME)LUA";
+
+            AZ::StringFunc::Replace(luaComponentBoilerplate, namePlaceholder, componentName.c_str());
+            return luaComponentBoilerplate;
+        }
+
+        void LuaEditorSystemComponent::MakeFilenameUnique(
+            const AZStd::string& directoryPath,
+            const AZStd::string& filename,
+            AZStd::string& outFullFilepath)
+        {
+            int fileCounter = 0;
+            while (AZ::IO::FileIOBase::GetInstance()->Exists(outFullFilepath.c_str()))
+            {
+                fileCounter++;
+                AZStd::string incrementalFilename = filename + AZStd::to_string(fileCounter);
```

Comment:
Nit: could be `const`.

Reply:
Done.

Reply:
If it's the function you meant, then done! :)

Reply:
I `const` the string as well! Is `const` a verb? Had a chat with Danilo about how `const` on local variables can improve code clarity and maintenance. I can see how it could also be confusing to astute readers though, if you forget to `const` something that could have been `const`. Seems like a small price to pay overall though.